src/lib/components/SearchBar/SearchBar.js

```diff
@@ -31,33 +31,54 @@ class SearchBar extends Component {
     this.updateQueryString(this.state.currentValue);
   };
 
+  onBtnSearchClick = (event, input) => {
+    this.executeSearch();
+  };
+
+  onKeyPress = (event, input) => {
+    if (event.key === 'Enter') {
+      this.executeSearch();
+    }
+  };
+
   render() {
     const { placeholder, overridableId } = this.props;
     return (
       <Element
-        placeholder={placeholder}
-        queryString={this.state.currentValue}
+        onBtnSearchClick={this.onBtnSearchClick}
```

Comment:
I am bit lost here, shouldn't you use the prop?
```suggestion
        const { placeholder, overridableId, ..., onBtnSearchClick, ... } = this.props;
        onBtnSearchClick={onBtnSearchClick}
```
same for other props.